src/components/Renderer.js

```diff
@@ -0,0 +1,1060 @@
+      .classed('axies', true)
+      .attr('class', 'axis');
+
+    // setup the axes
+    timelineScale = d3.time.scale()
+      .range([0, CHART_WIDTH - 1])
+      .domain([new Date(1900, 12, 15), new Date()]);
+
+    timelineAxis = d3.svg.axis()
+      .scale(timelineScale).tickSize(-20, -10, 0)
+      .tickSubdivide(true);
+
+    axiesGroup.call(timelineAxis);
+
+    // create the fdl instance
+    force = d3.layout.force()
+      .gravity(GRAVITY)
+      .linkStrength(LINK_STRENGHT)
+      .charge(function(d) {
+        return d.getProperty('hidden')
+          ? -CHARGE_HIDDEN
+          : -(Math.random() * CHARGE_RANDOM + CHARGE_BASE);})
+      .linkDistance(function(link) {
+        var base = LINK_BASE;
+
+        if (link.source == centerPerson || link.target == centerPerson)
+          base = NODE_SIZE / 2 + LINK_MIN_OFFSET;
+        else
+          base = NODE_SIZE / 4 + LINK_MIN_OFFSET;
+
+        return Math.random() * LINK_RANDOM + base;})
+      .size([CHART_WIDTH, CHART_HEIGHT]);
+
+    // handle window resize
+    d3.select(window)
+      .on('resize', function() {
+        const { width, height } = getViewportDimensions()
+
+        d3.select('#chartSVG')
+          .attr('height', height)
+          .attr('width', width)
+        
+        timelineScale.range([ 0, width - 1])
+
+        timelineAxis = d3.svg.axis()
+          .scale(timelineScale)
+          .tickSize(-20, -10, 0)
+          .tickSubdivide(true)
+        
+        force
+          .size([ width, height ])
+
+        axiesGroup.call(timelineAxis)
+      })
+
+    resolve(svg, force)
+  })
+}
+
+const render = function(history) {
+  $(document).ready(function() {
+    createChart().then((svg) => {
+      createSpecialData(function() {
+        var subject = establishInitialSubject();
+        querySubject(lengthen(subject, true), true, false, function () {
+          svg.selectAll('text.static-text')
+            .transition()
+            .duration(2000)
+            .style('fill', '#bbb');
+
+          svg.selectAll('text.loading')
+            .transition()
+            .style('fill', 'white')
+            .remove();
+        });
+
+        $(document).keydown(function(e){
+          switch (e.keyCode) {
+          case 37:
+            goBack();
+            break;
+          case 39:
+            goForward();
+            break;
+          }
+        });
+      })
+    })
+  })
+
+  function querySubject(subjectId, recordPast, recordFuture, callback) {
+    recordPast = recordPast || true
+    recordFuture = recordFuture || false
+
+    getPerson(subjectId, (graph) => {
+      if (graph.getNodes().length > 0) {
+        if (recordPast) savePast()
+        if (recordFuture) saveFuture()
+
+        centerPerson = graph.getNode(subjectId)
+        limitScreenNodes(graph)
+        updateChart(graph, force)
+        setWikiPage(centerPerson)
+      }
+
+      callback && callback()
+    })
+  }
+
+  function updateChart(graph, force) {
```

Comment:
This function is ripe for a refactor just because it's so freaking long at 319 lines, but don't do that now...